pyonfx/utils.py

```diff
@@ -143,68 +143,79 @@ class FrameUtility:
     You can use it to iterate over the frames going from ``start_ms``
     to ``end_ms`` and perform operations easily over multiple frames.
 
-    BACKGROUND KNOWLEDGE:
-
-    When reproducing a video+sub in a player, a line is rendered on the current frame
-    if the current time of the player is in between the line's start and end time.
-
-    The frame's duration can be determined by the fps value.
-
-    Let's now walk through an example: fps = 20 -> frame_duration = 50 ms.
-
-    The player will then seek for frames at the following times: 0, 50, 100, 150, ...
-
-    To accomodate this fact, frames' start and end times are positioned as follows:
-    - Frame 0: 0-25
-    - Frame 1: 25-75
-    - Frame 2: 75-125
-    - Frame 3: 125-175
-    - ...
-
-    This way, the frames' mid time will always be one of the player's seek times.
-    The only exception is the first frame, since we can't have negative times.
-
     Parameters:
         start_ms (positive int): Initial time in ms.
         end_ms (positive int): Final time in ms.
-        fps (positive int, float or Fraction, optional): Frames per second.
+        timestamps (ABCTimestamps): A timestamps object from [VideoTimestamps](https://github.com/moi15moi/VideoTimestamps/).
         n_fr (positive int, optional): Number of frames covered by each iteration.
 
     Returns:
         Returns a Generator yielding start_ms, end_ms, current frame index and total number of frames at each step.
 
     Example:
-        >>> FU = FrameUtility(0, 110, 20)
+        >>> # Let's assume to have an Ass object named "io" having a 20 fps video (i.e. frames are 50 ms long)
+        >>> FU = FrameUtility(0, 110, io.input_timestamps)
```

Comment:
If I remember correctly, you wanted to have a ``io.input_timestamps`` object.
Is it still needed?

Reply:
Hi Moi! Thank you so much for the work, I've jus seen the changes and they're looking really good!
Yes, I think it would be better to have it computed automatically and saved in the io object (we can also allow disabling it through a flag for optimization if wanted).

Once this is done, the PR is looking pretty good already, I'll just go through a read and we're ready!